controllers/namespace/init.sh.tmpl

```diff
@@ -23,89 +24,120 @@ im_nodes=(
 
-	echo "Downloading OneAgent package..."
-	if ! curl "${curl_params[@]}"; then
-		echo "Failed to download the OneAgent package."
-		exit "${fail_code}"
-	fi
+		if [[ "${proxy}" != "" ]]; then
+			curl_params+=("--proxy" "${proxy}")
+		fi
+
+		echo "Downloading OneAgent package..."
+		if ! curl "${curl_params[@]}"; then
+			echo "Failed to download the OneAgent package."
+			exit "${fail_code}"
+		fi
 
-	echo "Unpacking OneAgent package..."
-	if ! unzip -o -d "${bin_dir}" "${archive}"; then
-		echo "Failed to unpack the OneAgent package."
-		mv "${archive}" "${bin_dir}/package.zip"
-		exit "${fail_code}"
+		echo "Unpacking OneAgent package..."
+		if ! unzip -o -d "${bin_dir}" "${archive}"; then
+			echo "Failed to unpack the OneAgent package."
+			mv "${archive}" "${bin_dir}/package.zip"
+			exit "${fail_code}"
+		fi
+	else
+		printf "\Downloading OneAgent is not necessary, not in installer mode\n"
 	fi
-fi
+}
 
-echo "Configuring OneAgent..."
-echo -n "${INSTALLPATH}/agent/lib64/liboneagentproc.so" >> "${share_dir}/ld.so.preload"
+writeToConfFile() {
+	eval "echo ${1}" >> ${2}
+}
 
-for i in $(seq 1 $CONTAINERS_COUNT)
-do
-	container_name_var="CONTAINER_${i}_NAME"
-	container_image_var="CONTAINER_${i}_IMAGE"
+createConfigurationFiles() {
+	echo -n "${INSTALLPATH}/agent/lib64/liboneagentproc.so" >> "${share_dir}/ld.so.preload"
 
-	container_name="${!container_name_var}"
-	container_image="${!container_image_var}"
+	set +u # allow variable to be empty
+	host_tenant="${im_nodes[${K8S_NODE_NAME}]}"
+	set -u
 
-	container_conf_file="${share_dir}/container_${container_name}.conf"
+	for i in $(seq 1 $CONTAINERS_COUNT)
+	do
+		container_name_var="CONTAINER_${i}_NAME"
+		container_image_var="CONTAINER_${i}_IMAGE"
 
-	echo "Writing ${container_conf_file} file..."
-	echo "[container]
-containerName ${container_name}
-imageName ${container_image}
-k8s_fullpodname ${K8S_PODNAME}
-k8s_poduid ${K8S_PODUID}
-k8s_containername ${container_name}
-k8s_basepodname ${K8S_BASEPODNAME}
-k8s_namespace ${K8S_NAMESPACE}">>${container_conf_file}
+		container_name="${!container_name_var}"
+		container_image="${!container_image_var}"
 
-	if [[ ! -z "${host_tenant}" ]]; then
-		if [[ "{{.DynaKube.Status.ConnectionInfo.TenantUUID}}" == "${host_tenant}" ]]; then
-			echo "k8s_node_name ${K8S_NODE_NAME}
-k8s_cluster_id ${cluster_id}">>${container_conf_file}
-		fi
+		container_conf_file="${share_dir}/container_${container_name}.conf"
 
-	echo "
-[host]
-tenant ${host_tenant}">>${container_conf_file}
-	fi
-done
+		echo "Writing ${container_conf_file} file..."
+
+		writeToConfFile ${conf_base} ${container_conf_file}
+
+		if [[ "${host_tenant}" != "" ]]; then
+			if [[ "{{.DynaKube.Status.ConnectionInfo.TenantUUID}}" == "${host_tenant}" ]]; then
+				writeToConfFile ${conf_k8} ${container_conf_file}
+			fi
+			writeToConfFile ${conf_host} ${container_conf_file}
+		fi
+	done
+}
+
+####### MAIN #######
+printf "\nSetting fail code\n"
+setFailCode
+printf "\Downloading OneAgent\n"
```

Comment:
Maybe it's a nitpick, but I think a \n is missing.

Besides that everything looks fine to me.